src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsOutputImportStep.java

```diff
@@ -0,0 +1,100 @@
+package pl.edu.icm.pl.mxrdr.extension.workflow;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.harvard.iq.dataverse.DatasetFieldServiceBean;
+import edu.harvard.iq.dataverse.dataset.datasetversion.DatasetVersionServiceBean;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterFieldKey;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterRegistry;
+import edu.harvard.iq.dataverse.importer.metadata.MetadataImporter;
+import edu.harvard.iq.dataverse.importer.metadata.ResultField;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetField;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.workflow.WorkflowExecutionContext;
+import edu.harvard.iq.dataverse.workflow.step.Failure;
+import edu.harvard.iq.dataverse.workflow.step.FilesystemAccessingWorkflowStep;
+import edu.harvard.iq.dataverse.workflow.step.Success;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult.Source;
+import io.vavr.control.Try;
+import pl.edu.icm.pl.mxrdr.extension.importer.xds.XdsImporter;
+import pl.edu.icm.pl.mxrdr.extension.importer.xds.XdsImporterForm;
+
+public class XdsOutputImportStep extends FilesystemAccessingWorkflowStep {
+
+    private static final String WORKFLOW = "WORKFLOW";
+
+    static final String CORRECT_LP = "CORRECT.LP";
+    
+    ImporterRegistry importers;
+
+    private DatasetFieldServiceBean datasetFieldService;
+
+    private DatasetVersionServiceBean datasetVersionService;
+
+
+    public XdsOutputImportStep(Map<String, String> inputParams, DatasetFieldServiceBean datasetFieldService, DatasetVersionServiceBean datasetVersionService, ImporterRegistry importers) {
+        super(inputParams);
+        this.datasetFieldService = datasetFieldService;
+        this.datasetVersionService = datasetVersionService;
+        this.importers = importers;
+    }
+
+    @Override
+    public WorkflowStepResult resume(WorkflowExecutionContext context, Map<String, String> arg1, String arg2) {
+        return null;
+    }
+
+    @Override
+    public void rollback(WorkflowExecutionContext context, Failure failure) {
+    }
+
+    @Override
+    protected Source runInternal(WorkflowExecutionContext context, Path workdirPath) throws Exception {
```

Comment:
Too many abstraction levels in a single method - extract smaller methods with single abstraction level.